packages/shared_preferences/shared_preferences_web/example/integration_test/shared_preferences_web_test.dart

```diff
@@ -42,22 +42,30 @@ void main() {
       expect(await store.getAll(), isEmpty);
 
       html.window.localStorage['flutter.testKey'] = '"test value"';
-      html.window.localStorage['unprefixed_key'] = 'not a flutter value';
+      html.window.localStorage['unprefixed_key'] = '"not a flutter value"';
       final Map<String, Object> allData = await store.getAll();
       expect(allData, hasLength(1));
       expect(allData['flutter.testKey'], 'test value');
     });
 
+    testWidgets('getAllWithPrefix', (WidgetTester tester) async {
+      final SharedPreferencesPlugin store = SharedPreferencesPlugin();
+      const String prefix = 'prefix.';
+      expect(await store.getAllWithPrefix(prefix), isEmpty);
+
+      html.window.localStorage['${prefix}testKey'] = '"test value"';
+      html.window.localStorage['unprefixed_key'] = '"not a flutter value"';
+      final Map<String, Object> allData = await store.getAllWithPrefix(prefix);
+      expect(allData, hasLength(1));
+      expect(allData['${prefix}testKey'], 'test value');
+    });
+
```

Comment:
A new test with the same setup but a `''` prefix, asserting that it *can* read the non-Flutter-stored value, would be useful.